### DISCUSSION TITLE
Process for accessing device data in state

### DISCUSSION THREAD

**Discussion by davidjb** [home-assistant/architecture#583] [Entity Models]
At present, Device Tracker data exists inside Home Assistant's device registry, but this isn't (necessarily) accessible within state, such as in attributes, unless an integration makes it so.   This behaviour varies across and inside integrations, such as for [UniFi](https://www.home-assistant.io/integrations/unifi/), where this discussion began as part of the PR at https://github.com/home-assistant/core/pull/50440, which sought to make data usable in other areas of Home Assistant (sensors, templates, UI, etc).

That original PR is a change just for the UniFi platform for making infrastructure device attributes, such as make, model, MAC and software version available as state attributes, alongside other attributes obtained from the integration.  I'm most interested in the software version being available in state, as this is something that changes over time and something I've created Actions around; but the other data is useful as well - removing the need to hardcode values into actions, templates and UI.

A simple resolution to this could be to allow arbitrary state attributes for Device Tracker entities -- UniFi's implementation for infrastructure devices (UniFiDeviceTracker) already does this for `fan_level`, `overheating` and `upgradable`, and for a large number of client attributes (https://github.com/home-assistant/core/blob/922eec09098e0462b64bea91210a674512b8530a/homeassistant/components/unifi/device_tracker.py#L33-L56), some of which are considered static. If there's a concern over consistency of attribute keys across platforms, consts could be added.

Otherwise, the generalised ability to query the Device Registry for devices (or represent Registry devices as entities in state) so as to utilise this data (make/model/sw_version etc) within templates (sensors, actions, Lovelace etc) would also solve the issue.  This would be a much more massive change.

This discussion stems from [@MartinHjelmare raising concerns](https://github.com/home-assistant/core/pull/50440#discussion_r638524008_) about these data being static:

> The other info is static and not part of our entity state model for device trackers so that's not allowed to be added to state attributes.

However, as [I (@davidjb) pointed out](https://github.com/home-assistant/core/pull/50440#discussion_r638741852), the UniFi integration is already providing static attributes into state for tracked client devices, just not for infrastructure devices like routers.  So, that PR seems to be aligned with the current implementation.

**Comment by frenck** [comment]
This idea didn't gain enough traction in recent years and is running stale.

I'm going to close it for now; we can reconsider it in the future if this idea becomes relevant again.

../Frenck
